docs/specs/openapi.yaml

```
@@ -153,6 +79,314 @@ paths:
+      description: Fetches healthy status of specified checks.
+      tags:
+        - checks
+      parameters:
+        - name: level
+          in: query
+          description: Health check level.
+          schema:
+            type: string
+            enum: [alive, ready]
+        - name: names
+          in: query
+          description: The names of the checks to get. To get multiple checks, specify this parameter multiple times. If not set, get all checks.
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Check is healthy
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/GetHealth"
+              examples:
+                healthy:
+                  value:
+                    type: async
+                    status-code: 200
+                    status: OK
+                    result:
+                      healthy: true
+  /v1/checks:
+    get:
+      summary: Get checks
+      tags:
+        - checks
+      description: Fetch information about specific health checks (or all of them), ordered by check name.
+      parameters:
+        - name: level
+          in: query
+          description: Filter checks by level
+          schema:
+            type: string
+            enum: [alive, ready]
+        - name: names
+          in: query
+          description: The names of the checks to get. To get multiple checks, specify this parameter multiple times. If not set, get all checks.
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Get checks
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/GetChecks"
+              examples:
+                get checks:
+                  value:
+                    type: async
+                    status-code: 200
+                    status: OK
+                    result:
+                      - name: svc1-up
+                        status: up
+                        threshold: 3
+                        change-id: 91
+  /v1/changes:
+    get:
+      summary: Get changes
+      tags:
+        - changes
+      description: Fetch information for the specified changes.
+      parameters:
+        - name: select
+          in: query
+          description: Filter changes by status. Defaults to "in-progress".
+          schema:
+            type: string
+            enum: [all, in-progress, ready]
+            default: in-progress
+        - name: for
+          in: query
+          description: Filter changes for a specific service name.
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Get changes
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/GetChanges'
```

Should this be an array? From the example and also the code, it seems that it should be like this:
```suggestion
                $ref: '#/components/schemas/GetChanges'
```

---

Changes is a list, according to the code [here](https://github.com/canonical/pebble/blob/master/internals/daemon/api_changes.go#L163).

To double-check this, I did some tests. Run: `curl -X GET -H "Content-type: application/json" -H "Accept: application/json" "localhost:4000/v1/changes?select=all"` and we get a list as the result:

```json
{
    "type": "sync",
    "status-code": 200,
    "status": "OK",
    "result": [
        {
            "id": "1",
            "kind": "autostart",
            "summary": "Autostart service \"svc1\" and 1 more",
            "status": "Done",
            "tasks": [
                {
                    "id": "1",
                    "kind": "start",
                    "summary": "Start service \"svc1\"",
                    "status": "Done",
                    "progress": {
                        "label": "",
                        "done": 1,
                        "total": 1
                    },
                    "spawn-time": "2024-12-24T10:02:57.342163161+08:00",
                    "ready-time": "2024-12-24T10:02:58.349252454+08:00"
                },
                {
                    "id": "2",
                    "kind": "start",
                    "summary": "Start service \"svc2\"",
                    "status": "Done",
                    "progress": {
                        "label": "",
                        "done": 1,
                        "total": 1
                    },
                    "spawn-time": "2024-12-24T10:02:57.342167536+08:00",
                    "ready-time": "2024-12-24T10:02:58.352393079+08:00"
                }
            ],
            "ready": true,
            "spawn-time": "2024-12-24T10:02:57.342169703+08:00",
            "ready-time": "2024-12-24T10:02:58.352395537+08:00"
        },
        {
            "id": "2",
            "kind": "stop",
            "summary": "Stop all running services",
            "status": "Done",
            "tasks": [
                {
                    "id": "3",
                    "kind": "stop",
                    "summary": "Stop service \"svc1\"",
                    "status": "Done",
                    "progress": {
                        "label": "",
                        "done": 1,
                        "total": 1
                    },
                    "spawn-time": "2024-12-24T10:03:45.422085643+08:00",
                    "ready-time": "2024-12-24T10:03:45.456315268+08:00"
                },
                {
                    "id": "4",
                    "kind": "stop",
                    "summary": "Stop service \"svc2\"",
                    "status": "Done",
                    "progress": {
                        "label": "",
                        "done": 1,
                        "total": 1
                    },
                    "spawn-time": "2024-12-24T10:03:45.422093309+08:00",
                    "ready-time": "2024-12-24T10:03:45.453272143+08:00"
                }
            ],
            "ready": true,
            "spawn-time": "2024-12-24T10:03:45.425286893+08:00",
            "ready-time": "2024-12-24T10:03:45.456316351+08:00"
        }
    ]
}

The example in the spec is also in the same format, see the preview [here](https://canonical-pebble--543.com.readthedocs.build/en/543/reference/api/#/changes/get_v1_changes).

I think it's safe to resolve this comment for now, if I made a mistake please reopen.
```